client/driver/qemu.go

```diff
@@ -381,9 +431,26 @@ func (h *qemuHandle) Signal(s os.Signal) error {
 	return fmt.Errorf("Qemu driver can't send signals")
 }
 
-// TODO: allow a 'shutdown_command' that can be executed over a ssh connection
-// to the VM
 func (h *qemuHandle) Kill() error {
+	// If a graceful shutdown was requested at task start time,
+	// monitorPath will not be empty
+	if h.monitorPath != "" {
+		monitorSocket, err := net.Dial("unix", h.monitorPath)
+		if err == nil {
+			defer monitorSocket.Close()
+			h.logger.Printf("[DEBUG] driver.qemu - sending graceful shutdown command to qemu monitor socket at %s", h.monitorPath)
+			_, err = monitorSocket.Write([]byte(qemuGracefulShutdownMsg))
+			if err == nil {
```

Comment:
Is there any way to block until either the vm exits or the kill timeout is reached and make sure to always call `h.executor.ShutDown()` to kill Nomad's sidecar process?

I'm afraid this could leak VMs and Nomad executor processes.